python-lib/gluonts_forecasts/timeseries_preparation.py

```diff
@@ -99,48 +109,96 @@ def truncate_timestamps(self, df):
+        )
+
+    def _check_duplicate_dates(self, df, after_truncation=False):
+        """Check dataframe has no duplicate dates and raise an actionable error message """
+        duplicate_dates = self._count_duplicate_dates(df)
+        if duplicate_dates > 0:
+            error_message = f"Input dataset has {duplicate_dates} duplicate dates"
+            if after_truncation:
+                error_message += f" after truncation to '{self.frequency}' frequency. Please check the Frequency parameter."
+            else:
+                error_message += "."
+                if len(self.timeseries_identifiers_names) == 0:
+                    error_message += " Please check the Long format parameter."
+            raise ValueError(error_message)
 
-    def _has_duplicates_timestamps(self, df):
-        """Return True if there are no duplicate timestamps within each timeseries """
-        return any(df.duplicated(subset=self.timeseries_identifiers_names + [self.time_column_name], keep=False))
+    def _count_duplicate_dates(self, df):
+        """Return total number of duplicates dates within all timeseries """
+        return df.duplicated(subset=self.timeseries_identifiers_names + [self.time_column_name], keep=False).sum()
 
     def _check_timeseries_identifiers_columns_types(self, df):
         """ Raises ValueError if a timeseries identifiers column is not numerical or string """
         for column_name in self.timeseries_identifiers_names:
             if not is_numeric_dtype(df[column_name]) and not is_string_dtype(df[column_name]):
                 raise ValueError(f"Time series identifier '{column_name}' must be of string or numeric type")
 
+    def _check_no_missing_values(self, df):
+        for column_name in [self.time_column_name] + self.target_columns_names + self.timeseries_identifiers_names + self.external_features_columns_names:
+            if df[column_name].isnull().values.any():
+                raise ValueError(f"Column '{column_name}' must not have any missing values.")
+
+    def _log_timeseries_lengths(self, df, after_sampling=False):
+        """Log the number and sizes of time series and whether it's after sampling or not"""
+        if len(self.timeseries_identifiers_names) == 0:
+            timeseries_lengths = [len(df.index)]
+        else:
+            timeseries_lengths = list(df.groupby(self.timeseries_identifiers_names).size())
+        log_message = f"{len(timeseries_lengths)} time series"
+        if all(length == timeseries_lengths[0] for length in timeseries_lengths):
+            log_message += f" of {timeseries_lengths[0]} records"
+        else:
+            log_message += f" of {min(timeseries_lengths)} to {max(timeseries_lengths)} records"
+        if after_sampling:
+            logger.info(f"Sampled last records: train set contains {log_message}")
```

Comment:
Nitpick, it's not actually the train set (since the test will be removed) - my bad, it was my suggestions
```suggestion
            logger.info(f"Sampled last records to obtain {log_message}")
```

Reply:
@StanislasGuinel is that OK?